materials-data-facility/MDFConnectScala
# 0.1 Snapshot Release of MDF Connect Scala Bindings

# Problem 
As a Scala developer I want to be able to submit requests to MDF Connect without needing to code the raw REST calls.

# Approach
This library defines Scala bindings for the service. It includes Case Classes for Data Citations, as well as the enclosing Conversion Request message. It also has support for authenticating with Globus Auth and retrieving the token for the MDF Connect service.

# How to test
In order to test the library out you will need [SBT](https://www.scala-sbt.org) and [Scala 2.10](https://www.scala-lang.org/download/) installed.

Check out this branch. 

You can run the unit tests as
```bash
% set clean coverage test coverageReport
```

All of the tests will run and you can verify 96% of the code is covered by unit tests by examining the coverage report at `target/scala-2.10/scoverage-report/index.html`

The snapshot version of this library is available at 
[https://oss.sonatype.org/service/local/repositories/snapshots/content/org/materialsdatafacility/mdfconnect/mdfconnectscala_2.10/0.1-SNAPSHOT](https://oss.sonatype.org/service/local/repositories/snapshots/content/org/materialsdatafacility/mdfconnect/mdfconnectscala_2.10/0.1-SNAPSHOT)
